itunesSearch/itunesSearch/SearchResultController.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  SearchResultController.swift
+//  itunesSearch
+//
+//  Created by Bradley Yin on 8/6/19.
+//  Copyright © 2019 bradleyyin. All rights reserved.
+//
+
+import Foundation
+
+class SearchResultController {
+    let baseURL = URL(string: "https://itunes.apple.com/search?")!
+    
+    var searchResults: [SearchResult] = []
+    
+    func performSearch(searchTerm: String, resultType: ResultType, completion: @escaping (Error?)-> Void){
+        var components = URLComponents(url: baseURL, resolvingAgainstBaseURL: true)
+        let searchQueryItem = URLQueryItem(name: "term", value: searchTerm)
+        let searchQueryItem2 = URLQueryItem(name: "media", value: resultType.rawValue)
+        components?.queryItems = [searchQueryItem, searchQueryItem2]
+        
+        guard let requestURL = components?.url else{
+            NSLog("Request URL is nil")
+            completion(NSError())
+            return
+        }
+        print(requestURL)
+        var request = URLRequest(url: requestURL)
+        request.httpMethod = HTTPMethod.get.rawValue
+        URLSession.shared.dataTask(with: request) { (data, _, error) in
+            if let error = error {
+                print("Error fetching data: \(error)")
+                return
+            }
+            guard let data = data else {
+                print("No data returned from data task")
+                completion(NSError())
+                return
+            }
+            let decoder = JSONDecoder()
+            do{
```

Comment:
both here with `do{` and `}catch{` there's no space between the word and the brace. It's cleaner code to have a single space between.

`do {` and `} catch {`